### PR TITLE
Connect users to iCIMS account

### PR DESCRIPTION
**Pull request by tabfugnic**
* Implement an IcimsConnection model
* Users can connect iCIMS through the dashboard
* Clean up style violations throughout

### CODE REVIEW

File: app/views/dashboards/show.html.erb [hunk 1]
```diff
@@ -31,12 +29,22 @@
...
               <%= fa_icon "plug", text: t("dashboards.show.connect") %>
-            </a>
+            <% end %>
           </td>
         <% end %>
       </tr>
+
+      <tr class="icims-account">
+        <td>
+          <% unless @icims_connection.connected? %>
```
Comment (hrs) [inline app/views/dashboards/show.html.erb]: I favor `if !@icims_connection.connected?`.

**Comment by hrs** [comment]
Most of this looks fine to me (though, having been involved in its construction, I'm biased).

My big concern is that there's a lot of duplication with the Jobvite code. Should we extract that somehow? Maybe a common superclass for the controllers or something?

File: app/views/dashboards/show.html.erb (hunk 1 above)
Comment (tabfugnic) [inline app/views/dashboards/show.html.erb]: I just knew you would say that! My preference is `unless` but I can go either way. I assume the rest of the codebase probably wouldn't use `unless` so I will change it.

**Comment by cored** [comment]
@hrs One of the reasons I started experimenting with this PR: [41](https://github.com/namely/connect/pull/41) was because of what you are saying in your comment; I feel that there is no architecture define for Connect. Can you take a look at that PR too; so I can get some feedback?

File: app/views/dashboards/show.html.erb (hunk 1 above)
Comment (hrs) [inline app/views/dashboards/show.html.erb]: I guess an alternative would be `if @icims_connection.disconnected?`?

**Comment by tabfugnic** [comment]
@hrs I see your concern and I think that as soon as this PR gets in we will start to be able to abstract some of this stuff out. I can do that now if you'd like, but I personally would like to wait as more things shake out.

@cored I will take a look.

**Comment by hrs** [comment]
@tabfugnic Sure, I'm fine with giving things a little more time. Just gettin' the ball rolling on that conversation.

@cored I'll take a look, too!

**Comment by cored** [comment]
@tabfugnic @hrs thanks

**Comment by hrs** [comment]
Anyhoo, @tabfugnic, apart from that `unless`, I think this is good to merge.

**Pull request merged**